uint32 width
uint32 height
float64 x
float64 y
float64 yaw
bool redfind
bool apfind
bool first_find
